atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/InMemoryKeyValueService.java

```diff
@@ -338,7 +334,7 @@ private void putInternal(TableReference tableRef, Collection<Map.Entry<Cell, Val
                 // Save memory by sharing rows.
                 row = nextKey.row;
             }
-            byte[] oldContents = table.entries.putIfAbsent(new Key(row, col, timestamp), copy(contents));
+            byte[] oldContents = table.entries.putIfAbsent(new Key(row, col, timestamp), Arrays.copyOf(contents, contents.length));
```

Comment:
If you just want to abstract the length away, you can use `contents.clone()`. 

I don't have feelings either way, but there have been bikesheds built to house such differences in the past: http://markmail.org/message/5vyr3izg557h5u7o